didcomm/protocols/forward/forward.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+from typing import Optional, NamedTuple
+
+from didcomm.interfaces.did_resolver import DIDResolver
+from didcomm.interfaces.secrets_resolver import SecretsResolver
+from didcomm.types.algorithms import AnonCryptAlg
+from didcomm.types.message import Message
+from didcomm.types.mtc import MTC
+from didcomm.types.types import JSON, DID
+from didcomm.types.unpack_result import UnpackResult, Metadata
+
+
+class ForwardUnpackResult(NamedTuple):
+    forward_unpack_result: Optional[UnpackResult]
+    payload_unpack_result: Optional[UnpackResult]
+
+
+class Forwarder:
+
+    def __init__(self, secrets_resolver: SecretsResolver = None, did_resolver: DIDResolver = None):
+        pass
+
+    async def forward(self, packed_msg: JSON, to_did: DID,
+                      enc_alg: AnonCryptAlg) -> JSON:
+        return ""
+
+    async def unpack_forward(self, forward_msg: JSON) -> ForwardUnpackResult:
+        return ForwardUnpackResult(
+            forward_unpack_result=UnpackResult(
+                msg=Message(payload={}, id="", type=""),
+                metadata=Metadata(),
+                signed_payload=None
+            ),
+            payload_unpack_result=UnpackResult(
+                msg=Message(payload={}, id="", type=""),
+                metadata=Metadata(),
+                signed_payload=None
+            )
+
+        )
```

Comment:
Forward is something that we're still discussing (this is one of TBD items). 
The current proposal is the following (agree that it's not clear now, as docstrings are not attached yet, this is also TBD for this WIP PR):
- forward and core (pack/unpack) are separate packages. Forward is actually a protocol built on top of core pack/unpack.
- core contains `unpack` method that just unapcks a message. The message can be either Forward or a custom Message. `unpack` doesn't care.
- forward protocol contains additional helper method `unpack_forward` that may do unpack twice if it sees a message re-wrapped into forward ([link](https://identity.foundation/didcomm-messaging/spec/#rewrapping)).
This method can be called by Bob instead of a common `unpack` if he doesn't want to process unpack result each time to see if it's Forward or custom message, and just wants to always get the final custom message (with some optional metadata about forward details). BTW according to the link above, any message from a mediator may be re-wrapped with Forward, which means that any receiver may have to write the same boilerplate code as we provide within `unpack_forward`.

This proposal has a nice benefit separating core and forward stuff. But we are also considering another options: add a special flag `unpack_forward` to the `unpack` method itself, so that if this flag is true, `unpack` will return the final message (as `unpack_forward` message does in the current proposal).